### ISSUE TITLE
Unexpected test failure

### ISSUE THREAD

**Issue by Mubelotix** [meilisearch/meilisearch-rust#144]
The test `progress::test::test_wait_for_pending_updates_time_out` randomly fails in rare cases.

**Comment by sanders41** [comment]
Do you have any more information about the failure? I can’t recreate it on my end. Looking at the test my best guess is it finishes adding the documents before the time out. The timeout is currently set to 1 nanosecond so I guess it could be changed to 0? Since I can’t recreate the failure on my in I’m not sure how to test if that would make any difference.

**Comment by sanders41** [comment]
I have run some experiments and am still unsuccessful at getting the current test to fail. I found that I can make the test fail by setting the interval to 0 like `Some(Duration::from_nanos(0)), Some(Duration::from_nanos(1))`. I think this implies the random failures are indeed a timing issue?

I did test setting the timeout to 0 like `Some(Duration::from_millis(1)), Some(Duration::from_nanos(0))` and it passes, but because I couldn't get it to fail with this interval I don't know if it makes a difference in the random failure. @Mubelotix do you think it is worth you trying this since you were able to get the random failure?

**Comment by Mubelotix** [comment]
I think you are right about this. It is the last assertion that failed.
The thing is: the test did not timeout because Meilisearch returned a result in less than a millisecond. This should not have happened. So it is a Meilisearch bug, but the test should be able to handle it. 
Setting timeout to 0 should solve the problem as [the while condition](https://github.com/meilisearch/meilisearch-rust/blob/af3b363068815450e43231c45b423d8b3da7dc94/src/progress.rs#L127) will be 0 > 0, which is false. But it makes the test almost useless.

**Comment by sanders41** [comment]
Yes I agree with you about setting it to 0. However I can't think of any other solution so I'm not sure the best way to go with this.

**Comment by Mubelotix** [comment]
It happened right here on the CI while we were trying to merge #132. I can't reproduce it either.

**Comment by sanders41** [comment]
This takes the docker vs installed MeiliSearch out as a possible cause since CI is using docker also. Probably a good idea to try to log it. Currently its next to impossible to debug since we can't reproduce it.

**Comment by bidoubiwa** [comment]
Fixed!
